src/components/DatePicker/DatePicker.test.tsx

```diff
@@ -614,51 +614,47 @@ describe("DatePicker", () => {
       expect(disabled).toMatchSnapshot();
     });
 
-    /* // REVISIT THIS TEST
     it("should select two new dates", () => {
       render(
         <DatePicker
-          dateRange={true}
+          initialDate="3/2/1988"
+          initialDateTo="3/28/1988"
+          isDateRange
           labelText="Select the date range you want to visit NYPL"
         />
       );
       const fromInput = screen.getByLabelText(/From/i);
       const toInput = screen.getByLabelText(/To/i);
-      const date = getTodaysDateDisplay();
 
-      expect(screen.getAllByDisplayValue(date)).toHaveLength(2);
+      expect(fromInput).toHaveValue("1988-03-02");
+      expect(toInput).toHaveValue("1988-03-28");
+      // expect(screen.getAllByDisplayValue(date)).toHaveLength(2);
 
       // Let's select a new day.
       userEvent.click(fromInput);
       // The popup displays. Select a new day.
-      const newDateFrom =
-        todaysDate.getDate() > 28 ? 1 : todaysDate.getDate() + 1;
-      const newDateTo =
-        todaysDate.getDate() > 28 ? 18 : todaysDate.getDate() + 5;
-      const newDateToSelect = str_pad(newDateFrom);
-      userEvent.click(screen.getAllByText(newDateFrom)[0]);
+      const newDateFrom = 5;
+      const newDateTo = 25;
+      userEvent.click(screen.getByText(newDateFrom));
 
-      // We selected a new day but kept everything else the same. So we just
-      // need to remove the older day with the new "2" selected date.
-      // Example: 2021-08-01 -> 2 is selected -> 2021-08-02
-      const newFromValue = `${date.slice(0, -2)}${newDateToSelect}`;
-      expect(screen.getByDisplayValue(newFromValue)).toBeInTheDocument();
+      // We selected a new day but kept everything else the same.
+      // Example: 2021-03-02 -> 5 is selected -> 2021-03-05
+      expect(screen.getByDisplayValue("1988-03-05")).toBeInTheDocument();
       // The "To" input should only have the older value now.
-      expect(screen.getAllByDisplayValue(date)).toHaveLength(1);
+      // expect(screen.getAllByDisplayValue(date)).toHaveLength(1);
 
       // Now select the "To" date.
       userEvent.click(toInput);
       // The popup displays.
       userEvent.click(screen.getByText(newDateTo));
 
-      const newToValue = `${date.slice(0, -2)}${newDateTo}`;
-      expect(screen.getByDisplayValue(newToValue)).toBeInTheDocument();
-      // The original date value is no longer in display.
-      expect(screen.queryAllByDisplayValue(date)).toHaveLength(0);
+      expect(screen.getByDisplayValue("1988-03-25")).toBeInTheDocument();
+      // The original date values are no longer in display.
+      expect(screen.queryByDisplayValue("1988-03-02")).not.toBeInTheDocument();
+      expect(screen.queryByDisplayValue("1988-03-28")).not.toBeInTheDocument();
       // The "From" date value wasn't affected by this!
```

Comment:
I know I say this every time I look at your code and I'm going to keep saying it - it's so easy to read and parse your logic and it's really delightful.